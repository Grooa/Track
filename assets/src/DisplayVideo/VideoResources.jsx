import PropTypes from 'prop-types';

const VideoResources = ({ resources = null, className = '' }) => <section className={className}>
  <h2 className="colored">Resourecs</h2>

  <div className="data">
    {resources && <ul className="course-resources">
      {resources.map((r, key) => <li key={`resource-${key}`}>
        <a href={r.url} title={r.label} target="_blank">{r.label}</a>
      </li>)}
    </ul>}
  </div>
</section>;

VideoResources.propTypes = {
  className: PropTypes.string,
  resources: PropTypes.array,
};

export default VideoResources;
