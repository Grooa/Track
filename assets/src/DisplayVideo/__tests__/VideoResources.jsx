import { render, shallow } from 'enzyme';

import VideoResources from '../VideoResources';

describe('<VideoResources />', () => {
  let resourceList;
  let className;

  beforeEach(() => {
    resourceList = [
      {
        url: 'https://test.com/test.pdf',
        label: 'Some pdf resource',
      },
      {
        url: 'https://test.com/test2.pptx',
        label: 'Some PowerPoint resource',
      },
      {
        url: 'https://test.com/testTest',
        label: 'Some Website',
      },
    ];

    className = 'test-css';
  });

  it('should render empty list properly', () => {
    const wrapper = render(<VideoResources />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render list properly', () => {
    const wrapper = render(<VideoResources resources={resourceList} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should place className on <section>', () => {
    const wrapper = shallow(<VideoResources className={className} />);

    expect(wrapper.find('section').hasClass(className)).toBeTruthy();
  });
});
