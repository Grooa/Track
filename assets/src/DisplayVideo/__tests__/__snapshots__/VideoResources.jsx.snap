// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoResources /> should render empty list properly 1`] = `
<section
  class=""
>
  <h2
    class="colored"
  >
    Resourecs
  </h2>
  <div
    class="data"
  />
</section>
`;

exports[`<VideoResources /> should render list properly 1`] = `
<section
  class=""
>
  <h2
    class="colored"
  >
    Resourecs
  </h2>
  <div
    class="data"
  >
    <ul
      class="course-resources"
    >
      <li>
        <a
          href="https://test.com/test.pdf"
          target="_blank"
          title="Some pdf resource"
        >
          Some pdf resource
        </a>
      </li>
      <li>
        <a
          href="https://test.com/test2.pptx"
          target="_blank"
          title="Some PowerPoint resource"
        >
          Some PowerPoint resource
        </a>
      </li>
      <li>
        <a
          href="https://test.com/testTest"
          target="_blank"
          title="Some Website"
        >
          Some Website
        </a>
      </li>
    </ul>
  </div>
</section>
`;
